ZelBack/src/services/generalService.js

```diff
@@ -313,65 +298,30 @@ function splitRepoTag(repotag) {
-    throw new Error('Unable to communicate with Flux Services! Try again later.');
-  }
+  parseDockerTag(targetDockerTag);
 
-  const imageTags = resWhitelistRepo.data;
-  const pureOrganisations = [];
-  imageTags.forEach((imageTag) => { // todo revisit for more strict
-    const image = imageTag.substring(0, imageTag.lastIndexOf(':') > -1 ? imageTag.lastIndexOf(':') : imageTag.length);
-    if (image.includes('.')) {
-      if (image.split('/')[2]) { // abc.xyz/namespace/repository
-        const pureOrganisation = image.substring(0, image.lastIndexOf('/') > -1 ? image.lastIndexOf('/') : image.length); // or domain/namespace
-        pureOrganisations.push(pureOrganisation);
-      } else {
-        pureOrganisations.push(image);
-      }
-    } else {
-      const pureOrganisation = image.substring(0, image.lastIndexOf('/') > -1 ? image.lastIndexOf('/') : image.length); // or domain/namespace
-      pureOrganisations.push(pureOrganisation);
-    }
-  });
-
-  const splitRepoWhitelistAllowance = [];
-  if (splittedRepo.providerName === 'Docker Hub') {
-    splitRepoWhitelistAllowance.push(`${splittedRepo.namespace}/${splittedRepo.repository}:${splittedRepo.tag}`);
-    splitRepoWhitelistAllowance.push(`${splittedRepo.namespace}/${splittedRepo.repository}`);
-    splitRepoWhitelistAllowance.push(splittedRepo.namespace);
-    if (splittedRepo.namespace === 'library') {
-      splitRepoWhitelistAllowance.push(splittedRepo.repository);
-    }
-  } else if (splittedRepo.port) {
-    splitRepoWhitelistAllowance.push(`${splittedRepo.provider}:${splittedRepo.port}/${splittedRepo.namespace}`);
-    splitRepoWhitelistAllowance.push(`${splittedRepo.provider}:${splittedRepo.port}/${splittedRepo.namespace}/${splittedRepo.repository}`);
-    splitRepoWhitelistAllowance.push(`${splittedRepo.provider}:${splittedRepo.port}/${splittedRepo.namespace}/${splittedRepo.repository}/${splittedRepo.tag}`);
-  } else {
-    splitRepoWhitelistAllowance.push(`${splittedRepo.provider}/${splittedRepo.namespace}`);
-    splitRepoWhitelistAllowance.push(`${splittedRepo.provider}/${splittedRepo.namespace}/${splittedRepo.repository}`);
-    splitRepoWhitelistAllowance.push(`${splittedRepo.provider}/${splittedRepo.namespace}/${splittedRepo.repository}/${splittedRepo.tag}`);
+  // Cache?
+  const res = await serviceHelper.axiosGet(
+    'https://raw.githubusercontent.com/RunOnFlux/flux/master/helpers/repositories.json',
+  );
+
+  if (!res || !res.data) {
+    throw new Error(
+      'Unable to communicate with Flux Services! Try again later.',
+    );
   }
-  let isWhitelisted = false;
-  splitRepoWhitelistAllowance.forEach((allowedPart) => {
-    if (pureOrganisations.includes(allowedPart)) {
-      isWhitelisted = true;
-    }
-  });
 
-  if (!isWhitelisted) {
-    throw new Error('Repository is not whitelisted. Please contact Flux Team.');
+  const filteredImageTags = res.data.filter((imageTag) => targetDockerTag.slice(0, imageTag.length) === imageTag);
```

Comment:
This does not work well.
Imagine simple scenario where 'runonflux' namespace is whitelisted.
Now if you submit an app with 'runonfluxB/image:tag' it would work as runonflux is whitelisted. But it should not work as runonfluxB is not whitelisted.

Reply:
corrected